### ISSUE TITLE
Problems with fileuploads on "large" harddrives

### ISSUE THREAD

**Issue by jayminer81** [punktniklas/NiKom#14]
I can't upload files to my BBS, I get the error message:
"Tyvärr, gränsen för hur full disken får bli har överskridits!"

The disk I'm using is a 3GB disk and the partition NiKom is on is larger than 2GB.

I tried to set DISKFREE to 0 but it made no difference, and it doesn't matter if I make an area where I set the directory to my much smaller Workbench:-partition.

Here's the output from 'info'.

Mounted disks:
Unit	  Size	  Used	  Free Full Errs   Status   Name
PC0:      No disk present
RAM:     7690K    7690       0 100%   0  Read/Write Ram Disk
DH0:      102M   23912  185749  11%   0  Read/Write Workbench
DF0:      No disk present
DH1:    -1148634K   51745 6039596   1%   0  Read/Write Work

Volumes available:
Ram Disk [Mounted]
Work [Mounted]
Workbench [Mounted]

**Comment by patrikaxelsson** [comment]
NikTransfer.c:615 - bytesfree=(id->id_NumBlocks - id->id_NumBlocksUsed)*id->id_BytesPerBlock;
This will overflow if you have more than 2GB free space. Between 2 and 4GB available space it will be negative and should always give this error unless DISKFREE is more negative.

A reasonable solution would be to calculating free blocks instead of Bytes and compare that with DISKFREE/BytesPerBlock. This would allow 1024GB partitions (with 512Bytes BytesPerBlock).

**Comment by punktniklas** [comment]
Interesting :-)

So, yes, the problem is exactly as @patrikaxelsson points out, the 32 bit variable bytesfree overflows and wraps into negative values.

It seems like the info command is having similar problems? Are there other places where you run into problems? I'm not up to date with AmigaDos issues in the 21st century..

Anyway the fix is easy. We should probably introduce that "0" means don't check at all as well.

As long as the number of bytes free on your partition is in the 2-4GB range you should be able to work around the problem by configuring DISKFREE to INT_MIN (-2147483648) (There is no check in the code that reads the config that will protest against this value.)

How urgent is a fix for you? Would you need a fix to 1.61 (1.62?) or can you wait for 2.0? Or would you prefer running a development version of 2.0 when the fix is in?

Also, out of curiosity.. How do you upload? Are you using an old school terminal program over a serial line? Or are you using a shell/terminal that supports ZModem?

**Comment by punktniklas** [comment]
And, I forgot, the reason why it doesn't help to create a file area on a different partition is because it's not the partition of the file area that is checked. It's the partition of the temp upload directory. (You could argue that it's a bit stupid that only this partition is checked, but that's how it currently works..)

**Comment by jayminer81** [comment]
Well there are many cases where having a "big" drive can cause problems in AmigaOS, but there are also workarounds for most of them (new scsi.device, newer versions of programs, newer filesystems...). I'm running plain AmigaOS 3.1 without many patches on this machine though.

A fix is not that urgent to me, files on the BBS doesn't exactly feel all that important nowadays :) and the workaround I'm using at the moment is that I filled the disk up with a big file to get less than 2GB free. 

I found another issue with file-uploads though, but I will make another issue for that :/
